<launch>

  <!-- Load hand geometry parameters -->
  <include file="$(find sli_gpd)/launch/hand_geometry.launch">
    <arg name="node" value="generate_candidates_sli" />
  </include>

	<node name="generate_candidates_sli" pkg="sli_gpd" type="generate_candidates_sli" output="screen">

    <!-- Where the point cloud is stored -->
    <param name="cloud_file_name" value="/homeL/demo/ws_grasp/src/ros_gpg/data/krylon.pcd" />
    <!-- Plotting parameters -->

    <!-- Preprocessing of point cloud -->
    <param name="voxelize" value="true"/> <!-- Voxelize the point cloud -->
    <param name="remove_outliers" value="false"/> <!-- Remove statistical outliers from the point cloud -->
    <rosparam param="workspace"> [-1, 1, -1, 1, -1, 1] </rosparam> <!-- (x,y,z)-dimensions of the workspace -->
    <rosparam param="camera_position"> [0, 0, 0] </rosparam> <!-- The position of the camera -->

    <!-- General parameters -->
    <param name="num_samples" value="100" /> <!-- The number of samples to draw from the point cloud -->
    <param name="num_threads" value="4" /> <!-- The number of CPU threads to be used -->

    <!-- Parameters for local grasp candidate search -->
    <param name="nn_radius" value="0.01" /> <!-- Radius for the point neighborhood search -->
    <param name="num_orientations" value="8" /> <!-- Number of orientations to consider -->

    <!-- Filtering of grasp candidates -->
    <param name="downward_filter" value="true"/>
	</node>
</launch>
